### PR TITLE
Major refactor

### PR DESCRIPTION
**Pull request by sansyrox**
## Major Refactor! Needs Help!

Todo

- [x] Convert Classes to Function
- [x] Add the original landing page to this version
- [ ] Move out Component from Video.js render method to individual file.
- [x] Convert CSS to Scss files
- [ ] Move the ConnectToSocketServer function in Video.js file to a separate utils folder.
- [ ] Add Recoil

Now, the server has been separated out to a different repo.

To run the server, git clone this repo: https://github.com/mexili/incogly-backend

and use `yarn dev` command in both the repos

you all can fork the branch `major_refactor` and make the prs to this branch only.

All the issues will be labelled as hard as refactors are harder in general.

### CODE REVIEW

**Comment by sansyrox** [comment]
@pranay-simejia , this is the refactor PR.

**Comment by NishantJawla** [comment]
@sansyrox can I convert roompage.js from class-based component to a functional-based component

**Comment by sansyrox** [comment]
@NishantJawla , if you take a look at this Pr, the room page.js file has been removed.

we have two class components here, we need to change them

**Comment by Jassi10000-zz** [comment]
I work to work on #69 , pls assign me
@sansyrox sir I have mentioned the issue assignment here

**Comment by nimit2801** [comment]
@sansyrox is the Todo here our issue list for this PR?

**Comment by sansyrox** [comment]
@nimit2801 , yes

**Comment by sansyrox** [comment]
@Jassi10000 , that is not an active issue anymore. All the issues that we are focussing on atm are present in this PR's description.

**Comment by Jassi10000-zz** [comment]
> @Jassi10000 , that is not an active issue anymore. All the issues that we are focussing on atm are present in this PR's description.

This thing I didn't knew as this was opened in the issues list , ok close it then , so that others also don't get confused

**Comment by sansyrox** [comment]
> This thing I didn't knew as this was opened in the issues list , ok close it then , so that others also don't get confused

No worries 😄  Would still love your contributions here.

**Comment by NishantJawla** [comment]
@sansyrox  I have opened a pull request regarding the CSS to scss file conversion please see it

**Comment by sansyrox** [comment]
> @sansyrox I have opened a pull request regarding the CSS to scss file conversion please see it

Merged!

**Comment by nimit2801** [comment]
hey, @sansyrox do we've to convert Class in Video.js and its methods to functional components?

**Comment by sansyrox** [comment]
Yes @nimit2801

**Review by Jassi10000-zz — COMMENTED** [review]
looks good

**Comment by sansyrox** [comment]
@Jassi10000 
> looks good

It certainly doesn't. Please don't write blind reviews.
Thanks!

**Comment by Jassi10000-zz** [comment]
> @Jassi10000
> 
> > looks good
> 
> It certainly doesn't. Please don't write blind reviews.
> Thanks!

I just commented on the code that was changed to functional component , that was looking fine to me

File: src/components/Video/index_old.jsx [hunk 1]
```diff
@@ -0,0 +1,793 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
```
Comment (sansyrox) [inline src/components/Video/index_old.jsx:1]: @raghavdhingra , do we need this index_old.jsx?

Comment (raghavdhingra) [reply src/components/Video/index_old.jsx:1]: If index.jsx file is tested thoroughly then we can delete the old jsx file @sansyrox

**Comment by sansyrox** [comment]
Makes sense. Let us deploy the server and try to have the next convo on
incogly only. 🔥

**Comment by raghavdhingra** [comment]
Definitely 💪

**Comment by sansyrox** [comment]
closing pr.

Made it the default

**Pull request closed without merging**